src/gcp/cloudfunctionsv2.ts

```diff
@@ -510,86 +511,80 @@ export function functionFromEndpoint(cloudFunction: backend.FunctionSpec, source
+  proto.renameIfPresent(gcfFunction.serviceConfig, endpoint, "maxInstanceCount", "maxInstances");
 
-  if (backend.isEventTrigger(cloudFunction.trigger)) {
+  if (backend.isEventTriggered(endpoint)) {
     gcfFunction.eventTrigger = {
-      eventType: cloudFunction.trigger.eventType,
+      eventType: endpoint.eventTrigger.eventType,
     };
     if (gcfFunction.eventTrigger.eventType === PUBSUB_PUBLISH_EVENT) {
-      gcfFunction.eventTrigger.pubsubTopic = cloudFunction.trigger.eventFilters.resource;
+      gcfFunction.eventTrigger.pubsubTopic = endpoint.eventTrigger.eventFilters.resource;
     } else {
       gcfFunction.eventTrigger.eventFilters = [];
-      for (const [attribute, value] of Object.entries(cloudFunction.trigger.eventFilters)) {
+      for (const [attribute, value] of Object.entries(endpoint.eventTrigger.eventFilters)) {
         gcfFunction.eventTrigger.eventFilters.push({ attribute, value });
       }
     }
 
-    if (cloudFunction.trigger.retry) {
-      logger.warn("Cannot set a retry policy on Cloud Function", cloudFunction.id);
+    if (endpoint.eventTrigger.retry) {
+      logger.warn("Cannot set a retry policy on Cloud Function", endpoint.id);
     }
-  } else if (cloudFunction.trigger.allowInsecure) {
-    logger.warn("Cannot enable insecure traffic for Cloud Function", cloudFunction.id);
+  } else if (backend.isScheduleTriggered(endpoint)) {
+    // trigger type defaults to HTTPS.
+    gcfFunction.labels = { ...gcfFunction.labels, ["deployment-scheduled"]: "true" };
   }
-  proto.copyIfPresent(gcfFunction, cloudFunction, "labels");
 
   return gcfFunction;
 }
 
-export function endpointFromFunction(gcfFunction: CloudFunction): backend.FunctionSpec {
+export function endpointFromFunction(gcfFunction: CloudFunction): backend.Endpoint {
   const [, project, , region, , id] = gcfFunction.name.split("/");
-  let trigger: backend.EventTrigger | backend.HttpsTrigger;
-  if (gcfFunction.eventTrigger) {
+  let trigger: backend.Triggered;
+  if (gcfFunction.labels?.["deployment-scheduled"] === "true") {
     trigger = {
-      eventType: gcfFunction.eventTrigger!.eventType,
-      eventFilters: {},
-      retry: false,
+      scheduleTrigger: {},
+    };
+  } else if (gcfFunction.eventTrigger) {
+    trigger = {
+      eventTrigger: {
+        eventType: gcfFunction.eventTrigger!.eventType,
+        eventFilters: {},
+        retry: false,
+      },
     };
     if (gcfFunction.eventTrigger.pubsubTopic) {
-      trigger.eventFilters.resource = gcfFunction.eventTrigger.pubsubTopic;
+      trigger.eventTrigger.eventFilters.resource = gcfFunction.eventTrigger.pubsubTopic;
     } else {
       for (const { attribute, value } of gcfFunction.eventTrigger.eventFilters || []) {
-        trigger.eventFilters[attribute] = value;
+        trigger.eventTrigger.eventFilters[attribute] = value;
       }
     }
   } else {
-    trigger = {
-      allowInsecure: false,
-    };
+    trigger = { httpsTrigger: {} };
```

Comment:
q: what about `uri`?

tangent: does GCFv2 expose url for all functions types (event, http) like Cloud Run?

Reply:
URI is an attribute of Function, not httpsTrigger, and is set a few lines below (see line 584). It is present for all trigger types.